projects/plugins/crm/src/rest-api/v4/class-base.php

```diff
@@ -0,0 +1,197 @@
+<?php
+/**
+ * Jetpack CRM REST API base class.
+ *
+ * @package Automattic\Jetpack_CRM\REST_API
+ */
+
+namespace Automattic\Jetpack_CRM\REST_API\V4;
+
+use Automattic\Jetpack_CRM\REST_API\Util\Authentication;
+use Automattic\Jetpack_CRM\REST_API\Util\Route_Scope;
+use WP_Error;
+
+/**
+ * Jetpack Partner API REST base class.
+ *
+ * @package Automattic\Jetpack_CRM\REST_API\V4
+ */
+abstract class Base {
```

Comment:
The `Base` name sounds quite generic. Maybe `API_Base` or `API_Controller` are more specific names.

Reply:
> although we use namespaces, then it still didn't provide a lot of context in the scope itself

Yes, sure, but my """complain""" is about the naming, `Base` is quite a neutral word (adjective). The question I had was, but a `base` of what? We could have many types of bases in this context: An API controller base, an endpoint base, even an exception base...